### PR TITLE
#260: Ark files backup

### PR DESCRIPTION
**Pull request by mdrlzy**
#260

### CODE REVIEW

File: app/src/main/java/space/taran/arknavigator/mvp/model/backup/ArkBackup.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import java.time.format.DateTimeFormatter
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
+import kotlin.io.path.Path
+import kotlin.io.path.createDirectories
+import kotlin.io.path.exists
+import kotlin.io.path.inputStream
+import kotlin.io.path.outputStream
+
+class ArkBackup(private val foldersRepo: FoldersRepo) {
```
Comment (kirillt) [inline app/src/main/java/space/taran/arknavigator/mvp/model/backup/ArkBackup.kt:20]: Maybe `StorageBackup`?

Comment (mdrlzy) [reply app/src/main/java/space/taran/arknavigator/mvp/model/backup/ArkBackup.kt]: `ArkBackup` also backup file with favorites. There may be other files in future.

Comment (kirillt) [reply app/src/main/java/space/taran/arknavigator/mvp/model/backup/ArkBackup.kt]: Actually "storage" is everything what is replicated/persisted here. We have favorites storage, tags storage and index.

Comment (mdrlzy) [reply app/src/main/java/space/taran/arknavigator/mvp/model/backup/ArkBackup.kt]: Changed to `StorageBackup`

**Review by kirillt — APPROVED** [review]

**Pull request merged**